### PR TITLE
add routing_tag_ids chosen select to batch form

### LINKED ISSUE #375: Ability to update routing tags for Dialpeers and Destination using batch update form
Routing->Dialpeers->Collection Actions->Update batch

image

refs #738 

chosen multi-select should be used

### PR DESCRIPTION
**Pull request by BigG1947**
closes #375

### CODE REVIEW

File: app/assets/javascripts/build_tags.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    
+    var tagChosen = $('select#routing_tag_ids.chosen');
+    var disabled = tagChosen.prop('disabled')
+
+    if (disabled) { 
+        $("[name='routing_tag_ids']").prop('disabled', true); 
+    }else if(!disabled && tagChosen.val() == ''){
+        $("[name='routing_tag_ids']").prop('disabled', false); 
+    }
+});
```
Comment (senid231) [inline app/assets/javascripts/build_tags.js:46]: 1. why did you add this handler on document instead of add it in upper function?
2. it looks like uglier duplicate of above code inside `tagsChosen.change({hidden: hidden}, function(event){`

**Review by senid231 — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/build_tags.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    tagsChosen.chosen({no_results_text: "No results matched", width: '240px', search_contains: true, allow_single_deselect: true});
+
+    tagsChosen.change({hidden: hidden}, function(event){
+        if ($(event.target).val() != ''){
+            event.data.hidden.prop('disabled', true);
+        }else{
+            event.data.hidden.prop('disabled', false);
+        }
+    });
```
[the remark this answers is not in the document]
Comment (BigG1947) [reply app/assets/javascripts/build_tags.js:32]: > not when tagsChosen changed

Also I need disable hidden field when user select some tags

File: app/assets/javascripts/build_tags.js (hunk 1 above)
Comment (BigG1947) [reply app/assets/javascripts/build_tags.js:46]: > 1. why did you add this handler on document instead of add it in upper function?
> 2. it looks like uglier duplicate of above code inside `tagsChosen.change({hidden: hidden}, function(event){`

This is not same event. This event triggered when user changed status of anyone checkbox.
When checkbox near tagChosen is not checked - hidden field must be disabled.
When checkbox is checked and tagChosen is not have any tags - hidden field must be enabled.

**Review by Fivell — APPROVED** [review]

File: spec/features/routing/destinations/batch_update_spec.rb [hunk 3]
```diff
@@ -163,6 +165,12 @@
       check :Short_calls_limit
       fill_in :short_calls_limit, with: assign_params[:short_calls_limit]
     end
+
+    if assign_params.key? :routing_tag_ids
+      check :Routing_tag_ids
+      routing_tags.select { |tag| assign_params[:routing_tag_ids].include? tag.id.to_s }
+                  .each { |tag| fill_in_chosen 'routing_tag_ids[]', with: tag.name, multiple: true, visible: false }
```
Comment (senid231) [inline spec/features/routing/destinations/batch_update_spec.rb:172]:
```
page.scroll_to find_field('routing_tag_ids[]')
assign_params[:routing_tag_ids].each do |tag_id|
  fill_in_chosen 'routing_tag_ids[]', with RoutingTag.find(tag_id).name, multiple: true
end
```

it will be more readable - and no need for `visible: false` here

File: spec/features/routing/destinations/batch_update_spec.rb [hunk 4]
```diff
@@ -183,5 +191,16 @@
         end.to have_enqueued_job(AsyncBatchUpdateJob).on_queue('batch_actions').with 'Routing::Destination', be_present, assign_params, be_present
       end
     end
+
+    context 'when routing tag ids field is empty' do
+      let(:assign_params) { { routing_tag_ids: '' } }
```
Comment (senid231) [inline spec/features/routing/destinations/batch_update_spec.rb:196]: value should be empty array

`let(:assign_params) { { routing_tag_ids: [] } }`

**Review by senid231 — CHANGES_REQUESTED** [review]
